File: Xamarin/SplashScreenX/SplashScreenX/SplashScreenX.Android/SplashScreen.cs
using Android.App;
using Android.Content;
using Android.Content.PM;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SplashScreenX.Droid
{
    [Activity(Label = "@string/app_name", Theme = "@style/MyTheme.Splash", Icon = "@drawable/splash_logo", MainLauncher = true, NoHistory = true)]
    public class SplashScreen : Activity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            // performs boot initialization
            base.OnCreate(savedInstanceState);
            // provides a cross-platform API
            Xamarin.Essentials.Platform.Init(this, savedInstanceState);
            // create a window in which to place the user interface
            SetContentView(Resource.Layout.activity_splash_screen);
        }

        protected override void OnResume()
        {
            // the activity is ready to start interacting with the user
            base.OnResume();
            // creation of the task object to manage an asynchronous method
            Task startupWork = new Task(() => { SimulateStartUp(); });
            startupWork.Start();
        }

        async void SimulateStartUp()
        {
            // stops the method for a defined time
            await Task.Delay(2000);
            // start a new activity
            StartActivity(new Intent(Application.Context, typeof(MainActivity)));
        }
    }
}

File: Android/SplashScreen/app/src/main/java/info/ribosoft/splashscreen/SplashScreen.java
package info.ribosoft.splashscreen;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

public class SplashScreen extends AppCompatActivity {
    int SPLAH_TIME_OUT = 3000;

    @Override
    // create the first activity
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);
        /* create a new handler to the message queue
           to be run after the specified ammount of time elapses */
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                // start a new activity
                startActivity(new Intent(SplashScreen.this, MainActivity.class));
                finish();
            }
        }, SPLAH_TIME_OUT);
    }
}